test/mocha/unhandled_rejections.js

```diff
@@ -677,6 +677,41 @@ if (windowDomEventSupported) {
 
         })
     });
+
+    if (typeof Worker !== "undefined") {
+        describe("dom events in a worker", function() {
+            var worker;
+            beforeEach(function () {
+                worker = new Worker("./worker.js");
+            });
+
+            afterEach(function () {
+                worker.terminate();
+            });
+
+            specify("are fired", function() {
+                return new Promise(function(resolve, reject) {
+                    var order = [];
+                    worker.onmessage = function (message) {
+                        switch(message.data) {
+                        case "unhandledrejection":
+                            order.push(1);
+                            break;
+                        case "rejectionhandled":
+                            order.push(2);
+                            resolve();
+                            break;
+                        default:
+                            throw new Error("unexpected message: " + message);
+                        }
+                    };
+
+                    worker.postMessage("reject");
+                }).timeout(500);
```

Comment:
I took the timeout from the similar test done outside of workers. I supposed there was some Bluebird-specific reason for a 500 figure but I did not investigate. And I supposed that if the 500 figure in other tests did not cause problems there then it would not cause problems here either. This is a conservative stance. If I were to decide to reduce the timeout because it works for me when I try it, there's no telling whether I'd be breaking things elsewhere (e.g. if the suite is run on a platform where a timeout lower than 500 would lead to false failures).

Same reason I did not introduce arrows in this patch, by the way.